core/src/main/java/fj/P2.java

```diff
@@ -1,8 +1,10 @@
 package fj;
 
 import static fj.Function.*;
-
+import static fj.data.optic.PLens.pLens;
```

Comment:
I'm concerned about putting optic instances in the same classes. The reason is that it will cause cyclic references between packages and classes (e.g. PLens depends on List and List depends on PLens). In Monocle there is no such problem: optic instances are located in monocle.std package. What if we use the same strategy and create fj.data.optic.std package?

Also, what do you think about creating a new artifact functionaljava-lens with all fj.data.optic stuff in it?